fix(combobox): add chevron icon at end of input regardless of selecti…

y0n4 opened the pull request:
…on mode (#3055)

**Related Issue:** #

## Summary

<!--

Please make sure the PR title and/or commit message adheres to the https://www.conventionalcommits.org/en/v1.0.0/ specification.

Note: If your PR only has one commit and it is NOT semantic, you will need to either

a. add another commit and wait for the check to update
b. proceed to squash merge, but make sure the commit message is the same as the title.

This is because of the way GitHub handles single-commit squash merges (see https://github.com/zeke/semantic-pull-requests/issues/17)

If this is component-related, please verify that:

- [ ] feature or fix has a corresponding test
- [ ] changes have been tested with demo page in Edge

---

If this is skipping an unstable test:

- include info about the test failure
- submit an unstable-test issue by [choosing](https://github.com/Esri/calcite-components/issues/new/choose) the unstable test template and filling it out

-->

jcfranco reviewed (approved):
LGTM! 

@macandcheese can you take a glance at this one? 👀

macandcheese reviewed (approved):
Looks good 👍

macandcheese commented:
I think this will need to pull in master to integrate the changes made to combobox height here: https://github.com/Esri/calcite-components/pull/3094

y0n4 commented:
current branch is now up-to-date with master branch

macandcheese commented:
Thanks - looks like the only issue is the misaligned caret with the selection-mode is multi and there are chips appended: 
![Screen Shot 2021-10-01 at 3 38 28 PM](https://user-images.githubusercontent.com/4733155/135693193-3b7e3570-c054-4cf7-a1eb-3bd4e8da18f5.png)

y0n4 commented:
Updated to have chevron icon horizontally centered. I have also fixed the multi-select mode chip padding to be more balanced

Before- all the padding is focused on top which makes the children within uneven and chevron icon isn't exactly centered
<img width="292" alt="Screen Shot 2021-10-06 at 12 07 18 PM" src="https://user-images.githubusercontent.com/25360903/136271978-65fac95d-d3c3-4cd6-bc6a-508586da6c3c.png">

After - padding is evenly distributed and does not affect the chevron icon
<img width="290" alt="Screen Shot 2021-10-06 at 11 50 05 AM" src="https://user-images.githubusercontent.com/25360903/136272011-8639c5e5-ba49-4475-9a13-9358b969a3bf.png">
